### PR TITLE
[AI] API for Prompt Guard webhook

### PR DESCRIPTION
**Pull request by EItanya**
# Description

## API changes

- Fix the name of `PromptGuard` to be spelled correctly
- Add `Webhook` field to request and response `webhook`
- Make explicit `Regex` field for request/response `PromptGuard`

## Code changes

None

## CI changes

None

## Docs changes

None

# Context

Users want to be able to protect their AI requests in a flexible way.

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto [hunk 1]
```diff
@@ -508,39 +511,78 @@ message AIPromptEnrichment {
...
+    // A list of Regex patterns to match against the response.
+    // All matches will be masked before being sent back to the client.
+    // matches and builtins are additive.
     repeated string matches = 1;
+    // A list of built-in regexes to mask in the response.
+    // matches and builtins are additive.
+    repeated BuiltIn builtins = 2;
+  }
+
+  message Webhook {
```
[the remark this answers is not in the document]
Comment (shashankram) [reply projects/gloo/api/v1/enterprise/options/ai/ai.proto:544]: Just looking at the comment it isn't clear when and how this should be used. What webhook, when should it be used, example, etc., would help users.

**Review by shashankram — APPROVED** [review]

**Comment by EItanya** [comment]
@shashankram that's a good point, the code will be released with an openapi spec and then instructions will be added to the docs

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto [hunk 2]
```diff
@@ -424,7 +434,10 @@ message Embedding {
   }
 }
 
+// Settings for the Semantic Caching feature
 message SemanticCache {
+
+  // Settings for the Redis database
   message Redis {
```
Comment (npolshakova) [inline projects/gloo/api/v1/enterprise/options/ai/ai.proto:441]: Should redis/weaviate be a oneof? Can you configure both?

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto [hunk 3]
```diff
@@ -508,39 +520,88 @@ message AIPromptEnrichment {
...
+    // matches and builtins are additive.
+    repeated BuiltIn builtins = 2;
+  }
+
+  // Webhook settings for prompt guard
+  message Webhook {
+    // Host to send the traffic to.
+    string host = 1;
+    // Port to send the traffic to
+    uint32 port = 2;
```
Comment (npolshakova) [inline projects/gloo/api/v1/enterprise/options/ai/ai.proto:548]: Is the port optional or required?

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto (hunk 2 above)
Comment (EItanya) [reply projects/gloo/api/v1/enterprise/options/ai/ai.proto:441]: It's a `oneof` lower down in that message

**Review by npolshakova — APPROVED** [review]

**Pull request merged**